source/as-subtype.ts

```diff
@@ -0,0 +1,58 @@
+/**
+Cast the type of a variable to a subtype.
+
+This is useful in cases where the typescript engine is unable to catch certain implicit assertions.
```

Comment:
What cases? Can you describe succinctly and example situation where it might be needed?